connectors/sources/outlook.py

```diff
@@ -664,7 +664,9 @@ def __init__(self, configuration):
 
         self.tasks = 0
         self.queue = MemQueue(maxmemsize=QUEUE_MEM_SIZE, refresh_timeout=120)
-        self.fetchers = ConcurrentTasks(max_concurrency=MAX_CONCURRENCY)
+        self.fetchers = ConcurrentTasks(
+            max_concurrency=self.configuration["max_concurrent_tasks"]
```

Comment:
The new field is with `required: False`, so it can be empty even with `default_value`.

WDYT if we make it mandatory?

Reply:
> The new field is with required: False, so it can be empty even with default_value.

@wangch079  If user keep it empty, default value will be treated as max concurrent tasks. And I'd suggest to keep it optional.